apps/nfid-frontend/src/features/sdk/request-transfer/index.tsx

```diff
@@ -51,6 +52,7 @@ export const RequestTransfer: React.FC<IRequestTransferProps> = ({
   onConfirmIC,
 }) => {
   const [transferPromise, setTransferPromise] = useState<any>(undefined)
```

Comment:
why do you need transferPromise in state? you can show another component just onClick, and move complex logic to component which needs this promise